service/storage_service.cc

```diff
@@ -3608,7 +3608,7 @@ std::vector<gms::inet_address>
 storage_service::get_natural_endpoints(const sstring& keyspace,
         const sstring& cf, const sstring& key) const {
     sstables::key_view key_view = sstables::key_view(bytes_view(reinterpret_cast<const signed char*>(key.c_str()), key.size()));
-    dht::token token = dht::global_partitioner().get_token(key_view);
+    dht::token token = _db.local().find_schema(keyspace, key)->get_partitioner().get_token(key_view);
     return get_natural_endpoints(keyspace, token);
```

Comment:
s/key/cf/? did this pass tests?

Reply:
It did. Both unit at all level and BYO. Weird.

Reply:
I don't understand how... It seems to me like *any* use of get_natural_endpoints would have thrown on unfound table, no?

Reply:
Maybe we don't have tests that cover this. The only usage of this method I found was in api::storage_service maybe it's never called in tests.

Reply:
Fixed.